torch_frame/gbdt/tuned_xgboost.py

```diff
@@ -174,10 +177,15 @@ def objective(
 
         boost = xgboost.train(self.params, dtrain,
                               num_boost_round=num_boost_round,
-                              early_stopping_rounds=50, verbose_eval=False,
-                              evals=[(dvalid, 'validation')],
-                              callbacks=[pruning_callback])
-        pred = boost.predict(dvalid)
+                              early_stopping_rounds=early_stopping_rounds,
+                              verbose_eval=False, evals=[
+                                  (dvalid, 'validation')
+                              ], callbacks=[pruning_callback])
+        if boost.best_iteration:
+            iteration_range = (0, boost.best_iteration + 1)
+        else:
+            iteration_range = None
```

Comment:
will we ever enter this `else`?

Reply:
I think it's possible?
<img width="510" alt="image" src="https://github.com/user-attachments/assets/7fcbd68d-a7d6-4f9e-8787-e2caaebfe6e5">

Reply:
Yes, when there's no early stopping